crates/driver/src/boundary/liquidity/zeroex.rs

```diff
@@ -1,11 +1,78 @@
 use {
-    crate::{infra, infra::Ethereum},
+    crate::{
+        domain::liquidity::{
+            self,
+            zeroex::{self, Order},
+        },
+        infra::{self, Ethereum},
+    },
+    anyhow::anyhow,
     ethrpc::current_block::CurrentBlockStream,
-    shared::{http_client::HttpClientFactory, zeroex_api::DefaultZeroExApi},
-    solver::{liquidity::zeroex::ZeroExLiquidity, liquidity_collector::LiquidityCollecting},
+    model::order::OrderKind,
+    shared::{
+        http_client::HttpClientFactory,
+        price_estimation::gas::GAS_PER_ZEROEX_ORDER,
+        zeroex_api::DefaultZeroExApi,
+    },
+    solver::{
+        liquidity::{zeroex::ZeroExLiquidity, LimitOrder},
+        liquidity_collector::LiquidityCollecting,
+    },
     std::sync::Arc,
 };
 
+pub fn to_domain(
+    id: liquidity::Id,
+    limit_order: LimitOrder,
+) -> anyhow::Result<liquidity::Liquidity> {
+    let handler = limit_order
+        .settlement_handling
+        .as_any()
+        .downcast_ref::<solver::liquidity::zeroex::OrderSettlementHandler>()
```

Comment:
😬 hack alert. Can we add a comment in the code on why we are chosing this workaround and how a proper solution would look like?

Reply:
Done